src/DynamoCoreWpf/Utilities/WebView2Utilities.cs

```diff
@@ -15,31 +19,54 @@ public class DynamoWebView2 : WebView2
         private string tag;
         #endregion
 
+        Action<string> logger = System.Console.WriteLine;
+        private Task initTask = null;
+        private bool disposeCalled;
+
         public DynamoWebView2() : base()
         {
             tag = TestUtilities.WebView2Tag;
         }
 
+        /// <summary>
+        /// Wrapper over WebView2's EnsureCoreWebView2Async
+        /// Use this method instead of EnsureCoreWebView2Async
+        /// </summary>
+        /// <param name="logFn">Optional logging function. Will default to System.Console</param>
+        /// <returns></returns>
+        internal async Task Initialize(Action<string> logFn = null)
+        {
+            logger = logFn ?? logger;
+            initTask = EnsureCoreWebView2Async();
+            await initTask;
+
+            ObjectDisposedException.ThrowIf(disposeCalled, this);
+        }
+
         protected override void Dispose(bool disposing)
         {
+            if (disposeCalled) return;
+            disposeCalled = true;
+
             if (System.Environment.CurrentManagedThreadId != Dispatcher.Thread.ManagedThreadId)
             {
-                System.Console.WriteLine($"WebView2 instance with stamp {tag} is being disposed of on non-UI thread");
+                logger?.Invoke($"WebView2 instance with tag {tag} is being disposed of on non-UI thread");
             }
-            // We should dispose of webview2 only in the UI thread.
-            // Dispose can be called from the Finalizer (which can run on a non UI thread)
-            if (Dispatcher != null)
+
+            if (initTask != null && !initTask.IsCompleted)
```

Comment:
In line 41, you're awaiting `initTask`, so what will be the case where `initTask` is not complete?

Reply:
`await initTask` means that the execution of the function is put on hold until it finishes some internal awaits. Meanwhile the thread will continue to execute other function calls.
ex
```
DoStuff() {
      webview2 = new DynamoWebvView2();
      webview2.Initialize();// call but do not wait for it to finish
      webview2.Dispose();
}
``` This is pretty much what happens in some tests